infrastructure/smart-contracts/test/utils/getWinningKeyCountLocal.mjs

```diff
@@ -0,0 +1,56 @@
+import hardhat from "hardhat";
+const { ethers } = hardhat;
+
+export function getWinningKeyCountLocal(keyCount, boostFactor, bulkAddress, challengeId, confirmData, challengerSignedHash) {
+    if (keyCount <= 0) throw new Error("Error: Key Count Must Be Greater Than 0.");
+    if (boostFactor <= 0) throw new Error("Error: Boost factor must be greater than zero.");
+
+    const probability = boostFactor;
+
+    // Create a unique seed for this submission
+    const seedHash = ethers.keccak256(
+        ethers.solidityPacked(
+            ['address', 'uint256', 'bytes', 'bytes'],
+            [bulkAddress, challengeId, confirmData, challengerSignedHash]
+        )
+    );
```

Comment:
Did we confirm that the hash will be identical to the one from onchain ?

Reply:
Yes, I used the refereeCalculations.getWinningKeyCount function to compare results with the same inputs and they always matched.

Reply:
No I mean the hashes of the randomness. There might be something we overlook. Can you just "manually" print a log in the smart contract and call the function to your local node with a specific confirmData and signerhash and make sure its the same hash in the local calcualtion.

Reply:
I tried, but console log doesn't work in solidity for bytes.

As an alternative, I did the following:

Reply:
I fear that there are multiple ways the end values could mathc up without the hashes being identical if that makes sense.

I guyss it would be enough to confirm that the number representation of the random hash is the same, let me know what you think